FetchingEmailContents.cls

```diff
@@ -0,0 +1,106 @@
+                                 FROM Contact
+                                WHERE (FirstName = :contactName[0]
+                                  AND LastName = :contactName[1]) 
+                                  AND Email =:contactEmail];
+      try{
+            if (contacts.size() == 0)
+            {
+                Contact contactObj = new Contact();
+                contactObj.FirstName = ContactName[0];
+                contactObj.LastName = ContactName[1];
+                contactObj.Email = ContactEmail;
+                insert contactObj;
+                contactId = contactObj.id;
+            } 
+            else
+            {
+                contactId = contacts[0].id;
+            }
+         }
+        catch(DMLException e){} 
+            
+        
+        
+        
+        
+        try{
+               if(!String.isBlank(emailBody)){
+                
+               String emailBodyNew = emailBody.split('\n')[0];
+               List<Email_Handler__c> emailhandler = [SELECT Id
+                                                        FROM Email_Handler__c
+                                                       WHERE Email_Body__c = :emailBodyNew];
+              
+                if(emailhandler.size() != 0 )
+                { 
+                    emailHandlerid = emailhandler[0].id;
+                }
+                else
+                {
+                    
+                    Email_Handler__c newEmailHandler = new Email_Handler__c();
+                    newEmailHandler.Email_Subject__c =  emailSubject;
+                    newEmailHandler.Email_Body__c = emailBodyNew;
+                    insert newEmailHandler;
+                    emailHandlerid = newEmailHandler.id;
+                    
+                }
+                
+/*                Contact_Email_Handler_Junction__c newJunctionRecord = new Contact_Email_Handler_Junction__c();
+                newJunctionRecord.Contact_Name__c = contactId;
+                newJunctionRecord.Search_Query__c = emailHandlerid;
+                insert newJunctionRecord; */
+            } 
+            else{
+                Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+                String[] toAddress =new String[]{email.fromAddress};
+                mail.setToAddresses(toAddress);
+                mail.setSubject('Test subject');
+                mail.setplainTextBody('First line cannot be empty..');
+                Messaging.sendEmail(new Messaging.SingleEmailMessage[] { mail });
+            }
+            
+       }
+        catch(DMLException e){}
+        
+        System.debug('+++++++++++++++++++++++++++++++++++++++'+contactId);
+        System.debug('+++++++++++++++++++++++++++++++++++++++'+emailHandlerid);
+        
+        try{
+                Contact_Email_Handler_Junction__c newJunctionRecord = new Contact_Email_Handler_Junction__c();
+                newJunctionRecord.Contact_Name__c = contactId;
```

Comment:
Why there is no null check for `contactid` and `emailHandlerid`? Here you are trying to create record for null  values as well. Though it will not get created and throw exception. But this need to be handle by developer not by system.